lib/phoenix/socket.ex

```diff
@@ -47,6 +117,135 @@ defmodule Phoenix.Socket do
             topic: nil,
             transport: nil,
             transport_pid: nil
+
+
+  defmacro __using__(_) do
+    quote do
+      @behaviour Phoenix.Socket
+      import unquote(__MODULE__)
+      Module.register_attribute(__MODULE__, :phoenix_channels, accumulate: true)
+      @phoenix_transports %{}
+
+      transport :websocket, Phoenix.Transports.WebSocket,
+        serializer: Phoenix.Transports.JSONSerializer,
+        timeout: :infinity
+
+      transport :longpoll, Phoenix.Transports.LongPoll,
```

Comment:
To be clear, both transports will be explicit. We can generate them by default with the new user socket in new apps, long pooling will be commented.